src/server/v2.0/handler/project.go

```diff
@@ -209,6 +209,9 @@ func (a *projectAPI) CreateProject(ctx context.Context, params operation.CreateP
 		log.Warningf("failed to call JSONCopy on project metadata when CreateProject, error: %v", err)
 	}
 
+	// remove retention_id from p
+	delete(p.Metadata, "retention_id")
```

Comment:
How about move this logic to project controller? leaves the handler to do the authorization/validation(from API framework level) and  the let controller do the business logic(erase the retention ID).